### PR TITLE
Create PSA landing pages

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Create landing pages about the PSA Crypto API and about the driver interfaces. There isn't much content yet, but they will be updated as more becomes available. People have been asking for a page where they can check the status, so this is it.

### CODE REVIEW

File: docs/psa/accel/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# PSA cryptographic accelerator interface
+
+The accelerator interface lets you drivers for cryptographic accelerators into an implementation of the [PSA Cryptography API](../#application-programming-interface). Cryptographic accelerators perform cryptographic operations with keys in clear text.
```
Comment (Patater) [inline docs/psa/accel/index.md]:
```diff
- lets you drivers
+ lets you write drivers
```

Comment (Patater) [reply docs/psa/accel/index.md]:
```diff
- into an implementation of the
+ that interfaces with an implementation of the
```

Although I don't like saying "interface" twice in the sentence, the sentence as written doesn't make much sense to me.

File: docs/psa/entropy/index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# PSA entropy source driver interface
+
+The accelerator interface lets you drivers for entropy sources such as Hardware Random Number Generators (HRNG), also known as True Random Number Generators (TRNG), into an implementation of the [PSA Cryptography API](../#application-programming-interface).
```
Comment (Patater) [inline docs/psa/entropy/index.md]:
```diff
- The accelerator interface lets you drivers
+ The entropy source interface lets you write drivers
```

Comment (Patater) [inline docs/psa/entropy/index.md]:
```diff
- into an implementation of the
+ that interfaces with an implementation of the
```

File: docs/psa/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+### Accelerator driver interface
+
+The accelerator interface lets you drivers for cryptographic accelerators into an implementation of the PSA Cryptography API. Cryptographic accelerators perform cryptographic operations with keys in clear text.
+
+For more information, see [PSA cryptography accelerator driver interface](accel/).
+
+### Secure element driver interface
+
+The accelerator interface lets you drivers for external cryptoprocessors into an implementation of the PSA Cryptography API. External cryptoprocessors such as secure elements and smart cards perform cryptographic operations with keys accessed via opaque handles.
```
Comment (Patater) [inline docs/psa/index.md]:
```diff
- The accelerator interface lets you drivers
+ The secure element interface lets you write drivers
```

Comment (Patater) [inline docs/psa/index.md]:
```diff
- into an implementation of the
+ that interfaces with an implementation of the
```

File: docs/psa/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+For more information, see [PSA secure element driver interface](se/).
+
+### Entropy source driver interface
+
+The accelerator interface lets you drivers for entropy sources such as Hardware Random Number Generators (HRNG), also known as True Random Number Generators (TRNG), into an implementation of the PSA Cryptography API.
```
Comment (Patater) [inline docs/psa/index.md]:
```diff
- The accelerator interface
+ The entropy source interface lets you write drivers
```

Comment (Patater) [inline docs/psa/index.md]:
```diff
- into an implementation of the
+ that interfaces with an implementation of the
```

File: docs/psa/se/index.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# PSA secure element driver interface
+
+The accelerator interface lets you drivers for external cryptoprocessors into an implementation of the [PSA Cryptography API](../#application-programming-interface). External cryptoprocessors such as secure elements and smart cards perform cryptographic operations with keys accessed via opaque handles.
```
Comment (Patater) [inline docs/psa/se/index.md]:
```diff
- The accelerator interface lets you drivers
+ The secure element interface lets you write drivers
```

Comment (Patater) [inline docs/psa/se/index.md]:
```diff
- into an implementation of the
+ that interfaces with an implementation of the
```

File: docs/psa/accel/index.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<h1 id="psa-cryptographic-accelerator-interface">PSA cryptographic accelerator interface</h1>
```
Comment (Patater) [inline docs/psa/accel/index.html]: HTML in commit "Check in HTML versions" will need regeneration due to the `.md` updates.

File: docs/psa/index.md [hunk 7]
```diff
@@ -36,3 +36,9 @@ For more information, see [PSA secure element driver interface](se/).
 The accelerator interface lets you drivers for entropy sources such as Hardware Random Number Generators (HRNG), also known as True Random Number Generators (TRNG), into an implementation of the PSA Cryptography API.
 
 For more information, see [PSA entropy source driver interface](entropy/).
+
+## Feedback
+
+Arm welcomes feedback on the design of the PSA cryptography interfaces.
+If you think something could be improved, please open an [issue on the Mbed Crypto Github repository](https://github.com/ARMmbed/mbed-crypto/labels/api-spec).
```
Comment (Patater) [inline docs/psa/index.md]:
```diff
- Github
+ GitHub
```

File: docs/psa/accel/index.md (hunk 1 above)
Comment (gilles-peskine-arm) [reply docs/psa/accel/index.md]: I think I started to write “lets you plug … into …” then fiddled with the sentence quite a bit and it's still not quite right. Suggestions welcome. Maybe split this into two sentences?

Comment (Patater) [reply docs/psa/accel/index.md]: Yes, plug would make more sense. It wasn't the first word that came to mind when I read the first part of the sentence and tried to fill in the gap. Rewrite would be good.

File: docs/psa/index.md [hunk 8]
```diff
@@ -21,24 +21,27 @@ PSA includes functional specifications describing a hardware abstraction layer c
...
+The cryptographic accelerator driver interface lets you write drivers for hardware that performs cryptographic operations with keys in clear text.
+You can plug such drivers into any implementation of the PSA Cryptography API.
 
 For more information, see [PSA cryptography accelerator driver interface](accel/).
 
 ### Secure element driver interface
 
-The accelerator interface lets you drivers for external cryptoprocessors into an implementation of the PSA Cryptography API. External cryptoprocessors such as secure elements and smart cards perform cryptographic operations with keys accessed via opaque handles.
+The secure element driver interface lets you write drivers for external cryptoprocessors such as secure elements (SE), smart cards and hardware security modules (HSM) that perform operations on keys that never leave the external processor and are accessed only through opaque handles.
+You can plug such drivers into any implementation of the PSA Cryptography API.
```
Comment (Patater) [inline docs/psa/index.md]: Do you want another newline above here like in the other sections?

Comment (gilles-peskine-arm) [reply docs/psa/index.md]: I don't understand. A newline where? Are you talking about source formatting or rendering?

Comment (Patater) [reply docs/psa/index.md]: Source format.

Here, we have the following in source.

```
The secure element driver interface lets you write drivers for external cryptoprocessors such as secure elements (SE), smart cards and hardware security modules (HSM) that perform operations on keys that never leave the external processor and are accessed only through opaque handles.
You can plug such drivers into any implementation of the PSA Cryptography API.
```

In other similar places, we have:
Screenshot at 2019-10-16 09-29-23


However, I now see I was fooled by GitHub adding the vertical whitespace. There is no extra newline present.

**Pull request merged**